tests/subscriptions/test_worker.py

```diff
@@ -1,124 +1,133 @@
-                aggregations=[["count()", "", "count"]],
-                time_window=timedelta(minutes=500),
-                resolution=timedelta(minutes=1),
-            ),
-        )
-
-        now = datetime.utcnow()
-
-        timestamps = [
-            now,
-            (now + timedelta(minutes=self.minutes) + subscription.data.time_window),
-        ]
-
-        scheduler = DummyScheduler(
-            ScheduledTask(timestamp, subscription) for timestamp in timestamps
-        )
-
-        worker = SubscriptionWorker(
-            self.dataset,
-            ThreadPoolExecutor(),
-            {0: scheduler},
-            DummyProducer(broker),
-            result_topic,
-            DummyMetricsBackend(strict=True),
-        )
-
-        # Process the tick message and run the scheduled queries.
-        results = worker.process_message(
-            Message(
-                Partition(Topic("events"), 0),
-                0,
-                Tick(  # XXX: isn't actually used in the dummy scheduler
-                    offsets=Interval(0, 1),
-                    timestamps=Interval(timestamps[0], timestamps[-1]),
-                ),
-                timestamps[-1],
-            ),
-        )
-
-        assert results is not None
-
-        # Publish the results.
-        worker.flush_batch([results])
-
-        # Check to make sure the results were published.
-        # NOTE: This does not cover the ``SubscriptionTaskResultCodec``!
-        consumer: Consumer[SubscriptionTaskResult] = DummyConsumer(broker, "group")
-        consumer.subscribe([result_topic])
+from tests.base import dataset_manager
 
-        message = consumer.poll()
-        assert message is not None
-        assert message.partition.topic == result_topic
-        assert message.payload.task == results[0].task
-        assert message.payload.task.timestamp == timestamps[0]
-        assert message.payload.result == results[0].future.result()
-        request, result = message.payload.result
-        assert request.extensions["timeseries"] == {
-            "from_date": (timestamps[0] - subscription.data.time_window).isoformat(),
-            "to_date": timestamps[0].isoformat(),
-            "granularity": 3600,  # XXX: unused, no time grouping
-        }
-        assert result == {
-            "meta": [{"name": "count", "type": "UInt64"}],
-            "data": [{"count": 10}],
-        }
+
+class DummySubscriptionDataStore(SubscriptionDataStore):
+    def __init__(self) -> None:
+        self.__subscriptions: MutableMapping[UUID, SubscriptionData] = {}
+
+    def create(self, key: UUID, data: SubscriptionData) -> None:
+        self.__subscriptions[key] = data
+
+    def delete(self, key: UUID) -> None:
+        try:
+            del self.__subscriptions[key]
+        except KeyError:
+            pass
+
+    def all(self) -> Iterable[Tuple[UUID, SubscriptionData]]:
+        return [*self.__subscriptions.items()]
```

Comment:
This could possibly be moved into `sunbscriptions.store` if it's useful elsewhere. It would reduce the Redis dependency on a few other tests (the scheduler and the creator/deleter) so that might be a good thing to deal with later.